### PR TITLE
Framework for regularization losses

### PR DESCRIPTION
**Pull request by bstriner**
This provides some organization for collecting regularization losses.
- Any module can have an attribute `_losses`, which is a dictionary of names to losses
- `RegularizedLoss` collects all the losses from all the modules and adds it to the main loss.

Check regularized_mnist.py for an example of how it works. Code to use is relatively minimal. You get a nice tensorboard of all of the training and validation losses. (in combination with https://github.com/inferno-pytorch/inferno/pull/110)

Please let me know your thoughts on ways to approach this. I love the easy way TF organizes all your losses and inferno needs something similar. Tricky part is that if a criterion wants to create detailed logs then it needs access to the trainer and model. Tried to do it as cleanly as possible but open to suggestions.

### CODE REVIEW

**Comment by nasimrahaman** [comment]
Awesome!
